File: app/src/test/java/com/msaggik/algorithmsdatastructures/java/graph_tree/GraphAlgorithmsTest.java
package com.msaggik.algorithmsdatastructures.java.graph_tree;

import com.msaggik.algorithmsdatastructures.java.graph.graph_tree.BreadthFirstSearch;
import com.msaggik.algorithmsdatastructures.java.graph.graph_tree.DepthFirstSearch;
import com.msaggik.algorithmsdatastructures.java.graph.graph_tree.entity.GraphTree;

import org.junit.Assert;
import org.junit.Test;

public class GraphAlgorithmsTest {
    private final GraphTree tree
            = new GraphTree(3,
            new GraphTree(5,
                    new GraphTree(13,
                            new GraphTree(89),
                            new GraphTree(144,
                                    new GraphTree(4_181),
                                    new GraphTree(6_765,
                                            new GraphTree(10_946),
                                            new GraphTree(17_711)))),
                    new GraphTree(21,
                            new GraphTree(233),
                            new GraphTree(377))),
            new GraphTree(8,
                    new GraphTree(34,
                            new GraphTree(610),
                            new GraphTree(987)),
                    new GraphTree(55,
                            new GraphTree(1_597),
                            new GraphTree(2_584))));
    private final int sumGraphTree = 46_363;

    // DepthFirstSearchTest
    @Test(timeout = 100)
    public void testDepthRecursivelySumGraph() {
        int result = DepthFirstSearch.recursivelySumGraph(tree);
        Assert.assertEquals(result, sumGraphTree);
    }

    @Test(timeout = 100)
    public void testDepthIterativelySumGraph() {
        int result = DepthFirstSearch.iterativelySumGraph(tree);
        Assert.assertEquals(result, sumGraphTree);
    }

    // BreadthFirstSearchTest
    @Test(timeout = 100)
    public void testBreadthIterativelySumGraph() {
        int result = BreadthFirstSearch.iterativelySumGraph(tree);
        Assert.assertEquals(result, sumGraphTree);
    }
}

File: app/src/test/java/com/msaggik/algorithmsdatastructures/kotlin/graph_tree/GraphAlgorithmsTest.kt
package com.msaggik.algorithmsdatastructures.kotlin.graph_tree

import com.msaggik.algorithmsdatastructures.kotlin.graph.graph_tree.BreadthFirstSearch
import com.msaggik.algorithmsdatastructures.kotlin.graph.graph_tree.DepthFirstSearch
import com.msaggik.algorithmsdatastructures.kotlin.graph.graph_tree.entity.GraphTree
import org.junit.Assert
import org.junit.Test

class GraphAlgorithmsTest {
    private val tree: GraphTree = GraphTree(3,
        GraphTree(5,
            GraphTree(13,
                GraphTree(89),
                GraphTree(144,
                    GraphTree(4181),
                    GraphTree(6765,
                        GraphTree(10946),
                        GraphTree(17711)
                    )
                )
            ),
            GraphTree(21,
                GraphTree(233),
                GraphTree(377)
            )
        ),
        GraphTree(8,
            GraphTree(34,
                GraphTree(610),
                GraphTree(987)
            ),
            GraphTree(55,
                GraphTree(1597),
                GraphTree(2584)
            )
        )
    )
    private val sumGraphTree = 46363

    // DepthFirstSearchTest
    @Test(timeout = 100)
    fun testDepthRecursivelySumGraph() {
        val result = DepthFirstSearch.recursivelySumGraph(tree)
        Assert.assertEquals(result.toLong(), sumGraphTree.toLong())
    }

    @Test(timeout = 100)
    fun testDepthIterativelySumGraph() {
        val result = DepthFirstSearch.iterativelySumGraph(tree)
        Assert.assertEquals(result.toLong(), sumGraphTree.toLong())
    }

    // BreadthFirstSearchTest
    @Test(timeout = 100)
    fun testBreadthIterativelySumGraph() {
        val result = BreadthFirstSearch.iterativelySumGraph(tree)
        Assert.assertEquals(result.toLong(), sumGraphTree.toLong())
    }
}
